docs/middleware.md

```diff
@@ -0,0 +1,76 @@
+# Using Next 12 middleware on Netlify
+
+Next 12 introduces a new feature called [middleware](https://nextjs.org/docs/middleware), which are functions that are
+run before a request has finished processing, and can be used to modify the request or replace the response. For
+example, it can change headers, rewrite the request path, or return a different response entirely.
```

Comment:
Minor wordsmithing suggestions:

```suggestion
Next 12 introduces a new feature called [Middleware](https://nextjs.org/docs/middleware), in which functions
run before a request has finished processing. Middleware can be used to modify the request or replace the response. For
example, it can change headers, rewrite the request path, or return a different response entirely.
```